packages/core/src/browser/keybinding.ts

```diff
@@ -139,9 +143,20 @@ export class KeybindingRegistry {
         }
     }
 
-    registerKeybindings(...bindings: Keybinding[]): void {
+    protected doRegisterKeybindings(bindings: Keybinding[], scope: KeybindingScope = KeybindingScope.DEFAULT) {
         for (const binding of bindings) {
-            this.registerKeybinding(binding);
+            this.doRegisterKeybinding(binding, scope);
+        }
+    }
+
+    protected doRegisterKeybinding(binding: Keybinding, scope: KeybindingScope = KeybindingScope.DEFAULT) {
+        try {
+            if (!this.isKeybindingAvailable(binding)) { // Throws on parse error
```

Comment:
Should not we take the scope into the account?

Please add tests showing that it is possible to add keybindings to the user keymap which overrides default keybindings and does not fail.

Reply:
Well, I thought the scope was already taken into account, because most of this PR is just a refactoring (so that the binding availability check is shared between functions).

There is a concept of shadowing for a command, but shouldn't it consider keybinding shadowing, instead of command shadowing ??

Inside `isKeybindingShadowed` function:
https://github.com/marechal-p/theia/blob/81f5318bf32d27c03a7c1ddd33b21c4859e75867/packages/core/src/browser/keybinding.ts#L338

Reply:
To be sure about the use case:

1. Some key sequence (A) is bound to a command (B) by default.
2. User wants to map (A) to (C).
3. (A)->(B) still exists, but (A)->(C) has priority?

Reply:
from the user perspective, i would expect that my keybindings have priority and should overwrite defaults, so yes (A)->(C) has priority

cc @svenefftinge

Reply:
Yes, of course. The user keybindings override the application keybindings. 
Also, the 'key' for overriding a keybinding should be the key sequence not the command ID.

Reply:
My thoughts too, so registering KeyBindings do not need to check for collisions in different scopes, but only inside the same scope as theirs right ? In other words cross-scope collisions are not avoided, we rather check at execution time which scope has priority.

Reply:
yes, it looks like that it would be the easiest change with the current code